core/src/main/java/org/infinispan/CacheStream.java

```diff
@@ -318,6 +318,8 @@
    @Override
    <R1, A> R1 collect(Collector<? super R, A, R1> collector);
 
+   <R1, A> R1 collect(SerializableSupplier<Collector<? super R, A, R1>> supplier);
```

Comment:
Some Javadoc here would be great.

Reply:
Also do we want the non Serializable version override? In case if user wants to provide different serialization? This means just making this method default and calling the other.

Reply:
Why do you need to specify the accumulator type `A`? If you replace this with `?` the `Collectors::toList` in Will's comment below works.

Reply:
Replied below, for some reason the UI was not letting me reply here directly.